File: src/main/java/model/strategy/bono/SouthLookingDistanceProcessor.java
package model.strategy.bono;

import model.Coordinate;

public class SouthLookingDistanceProcessor extends DistanceProcessor
{
    public int getDistance(Coordinate actualCoordinate,
            Coordinate blockingCoordinate, Coordinate maxCoordinate)
    {
        int distance = 0;

        if (actualCoordinate.getY() > blockingCoordinate.getY()) {
            distance = getDistanceDirectly(actualCoordinate.getY(),
                    blockingCoordinate.getY());
        }
        if (actualCoordinate.getY() < blockingCoordinate.getY()) {
            distance = getDistanceClipped(actualCoordinate.getY(),
                    blockingCoordinate.getY(), maxCoordinate.getY());
        }

        return distance;
    }
}

File: snake-strategy-bono/src/main/kotlin/co/electric/snake/strategy/bonostrategy/distanceprocessor/SouthLookingDistanceProcessor.kt
package co.electric.snake.strategy.bonostrategy.distanceprocessor

import co.electric.snake.framework.model.Coordinate
import co.electric.snake.framework.model.Direction
import java.util.*

class SouthLookingDistanceProcessor : DistanceProcessor {

    override fun getDistance(direction: Direction, headCoordinate: Coordinate, blockingCoordinate: Coordinate, maxCoordinate: Coordinate): Optional<Int> {
        var distance: Optional<Int> = Optional.empty()
        if (direction == Direction.SOUTH) {
            val calculatedDistance: Int
            if (headCoordinate.y >= blockingCoordinate.y) {
                calculatedDistance = getDistanceDirectly(headCoordinate.y, blockingCoordinate.y)
            } else {
                calculatedDistance = getDistanceClipped(headCoordinate.y, blockingCoordinate.y, maxCoordinate.y)
            }
            distance = Optional.of(calculatedDistance)
        }
        return distance
    }

}
